src/main/java/pl/debuguj/parkingspacessystem/domain/ParkingSpace.java

```diff
@@ -0,0 +1,108 @@
+package pl.debuguj.parkingspacessystem.domain;
+
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import pl.debuguj.parkingspacessystem.controllers.ParkingSpaceController;
+import pl.debuguj.parkingspacessystem.enums.DriverType;
+import pl.debuguj.parkingspacessystem.exceptions.IncorrectEndDateException;
+
+import java.text.ParseException;
+import java.util.Date;
+
+/**
+ * Created by grzesiek on 07.10.17.
+ */
+public class ParkingSpace implements Space {
+
+    private static final Logger logger = LoggerFactory.getLogger(ParkingSpaceController.class);
+
+    private final String carRegistrationNumber;
+    private DriverType driverType = DriverType.REGULAR;
+    private Date beginDate;
+    private Date endDate;
+
+
+
+    public ParkingSpace(String carRegistrationNumber,
+                        Date beginDate,
+                        Date endDate) throws IncorrectEndDateException {
+        this.carRegistrationNumber = carRegistrationNumber;
+        this.beginDate = beginDate;
+        this.endDate = checkEndDate(endDate);
```

Comment:
co się stanie jak ktoś zmieni kolejność inicjalizacji pól?

Reply:
Ktoś? tzn kompilator podczas optymalizacji?. Słowo volatile na beginDate powinno pomóc. Jak uciekł mi brak synchronizacji przy zapisie danych to byłby chyba cud jakbym u siebie w kodzie to zauważył. Solidne review robicie.